element_deeplabcut/model.py

```diff
@@ -323,59 +323,56 @@ def insert_new_model(
             if dlc_config_fp.suffix in (".yml", ".yaml"):
                 with open(dlc_config_fp, "rb") as f:
                     dlc_config = yaml.safe_load(f)
+            if isinstance(params, dict):
+                dlc_config.update(params)
 
         # ---- Get and resolve project path ----
         project_path = find_full_path(
-            get_dlc_root_data_dir(), project_path or dlc_config["project_path"]
+            get_dlc_root_data_dir(), project_path or dlc_config.get("project_path")
         )
+        dlc_config["project_path"] = str(project_path)  # update if different
         root_dir = find_root_directory(get_dlc_root_data_dir(), project_path)
 
-        # ---- Build config ----
-        template_attributes = [
+        # ---- Verify config ----
+        needed_attributes = [
             "Task",
             "date",
-            "TrainingFraction",
             "iteration",
             "snapshotindex",
-            "batch_size",
-            "cropping",
-            "x1",
-            "x2",
-            "y1",
-            "y2",
-            "project_path",
+            "TrainingFraction",
         ]
-        config_template = {
-            k: v for k, v in dlc_config.items() if k in template_attributes
-        }
+        for attribute in needed_attributes:
+            assert (
+                attribute in dlc_config.keys()
+            ), f"Couldn't find {attribute} in config"

```

Comment:
It's better not to invoke `.keys()`

```suggestion
        for attribute in needed_attributes:
            assert attribute in dlc_config, f"Couldn't find {attribute} in config"
```